app/backend/app.py

```diff
@@ -83,10 +83,57 @@ async def assets(path):
     return await send_from_directory(Path(__file__).resolve().parent / "static" / "assets", path)
 
 
+# Decorator for routes that request a specific file that might require access control enforcement
+def authenticated_path(route_fn: Callable[[str], Any]):
+    async def auth_handler(path=""):
+        # If authentication is enabled, validate the user can access the file
+        auth_helper = current_app.config[CONFIG_AUTH_CLIENT]
+        search_client = current_app.config[CONFIG_SEARCH_CLIENT]
+        authorized = False
+        try:
+            auth_claims = await auth_helper.get_auth_claims_if_enabled(request.headers)
+            authorized = await auth_helper.check_path_auth(path, auth_claims, search_client)
+        except AuthError:
+            abort(403)
+        except Exception as error:
+            logging.exception("Problem checking path auth %s", error)
+            return error_response(error, route="/content")
+
+        if not authorized:
+            abort(403)
+
+        return await route_fn(path)
+
+    # Rename decorator to avoid view function mapping overwrite error
+    auth_handler.__name__ = route_fn.__name__
+
+    return auth_handler
+
+
+# Decorator for routes that might require access control. Unpacks Authorization header informaiton into an auth_claims dictionary
+def authenticated(route_fn: Callable[[Dict[str, Any]], Any]):
+    async def auth_handler():
+        auth_helper = current_app.config[CONFIG_AUTH_CLIENT]
+        try:
+            auth_claims = await auth_helper.get_auth_claims_if_enabled(request.headers)
+        except AuthError:
+            abort(403)
+
+        return await route_fn(auth_claims)
+
+    # Rename decorator to avoid view function mapping overwrite error
+    auth_handler.__name__ = route_fn.__name__
```

Comment:
The Flask documentation suggests that using functools.wraps() is the best practice here: https://flask.palletsprojects.com/en/3.0.x/patterns/viewdecorators/#login-required-decorator

Reply:
I was wondering why this was so weird, thanks for the suggestion!